plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.hdfs;
+
+import org.apache.hadoop.fs.FileContext;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+
+import java.io.IOException;
+
+public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+
+    @Override
+    protected BlobStore newBlobStore() throws IOException {
+        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
```

Comment:
Right, for example, in the case of S3, you would have to create a mock
implementation for the AmazonS3 interface.  Its a pretty lengthy interface,
but you would only need to provide mock implementations for the methods
that S3BlobStore uses, which are just a few.  The rest of the methods could
just return null or some dummy values.

On Sun, Jun 12, 2016 at 12:45 PM, gfyoung <notifications@github.com> wrote:

> In
> plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java
> <https://github.com/elastic/elasticsearch/pull/18815#discussion_r66723801>
> :
>
> > + * specific language governing permissions and limitations
> > + * under the License.
> > + */
> > +package org.elasticsearch.repositories.hdfs;
> > +
> > +import org.apache.hadoop.fs.FileContext;
> > +import org.elasticsearch.common.blobstore.BlobStore;
> > +import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
> > +
> > +import java.io.IOException;
> > +
> > +public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
> > +
> > +    @Override
> > +    protected BlobStore newBlobStore() throws IOException {
> > +        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
>
> Yeah, I was looking for some sort of mock objects as for the Google
> object, but I could not find any such thing. Should probably see if I can
> Mockito somehow.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/18815/files/5ecfabc0dffffcbec1cbf7c5c7db0e3ab05a8c07#r66723801>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABjkQVZ4QGKT05XXuqbFiz4DONPvQmbGks5qLDeUgaJpZM4Iy-tw>
> .
>